### PR TITLE
add support to filepath to create a PkPass

### LINKED ISSUE #4: Accept a `data` to create a PKPass instead of URL
I would like to create a `PkPass` using a `data` instead of a URL, do you accept a PR for this?

something like this:

```jsx
dispatch_async(dispatch_get_main_queue()) { () -> Void in
                    if response.errorMessage == nil
                    {
                        let pass = PKPass(data: response.serviceResponse , error: nil)
                        
                        let nextVC = PKAddPassesViewController(pass: pass)
                        self.navigationController?.presentViewController(nextVC, animated: true, completion: nil)
                    } else {
                        let alert : UIAlertView = UIAlertView(title: NSLocalizedString("warning", comment: ""),
                            message: response.errorMessage,
                            delegate: nil,
                            cancelButtonTitle: NSLocalizedString("ok", comment: "")
                        )
                        alert.show()
                    }
                    
                    VLoading.hideLoading()
                }
```

### PR DESCRIPTION
**Pull request by sibelius**
fix #4

### CODE REVIEW

**Comment by sibelius** [comment]
we'll change, tks

File: ios/RNWalletModule/RNWalletModule.m [hunk 1]
```diff
@@ -29,13 +29,22 @@ @implementation RNWalletModule
 }
 
 RCT_EXPORT_METHOD(
-	showAddPassController:(NSString *)pass
+                  showAddPassControllerFromData:(NSData *)data
```
Comment (erikpoort) [inline ios/RNWalletModule/RNWalletModule.m]: Doesn't seem as ReactNative supports NSData as a parameter:
https://facebook.github.io/react-native/docs/native-modules-ios.html#argument-types

Comment (sibelius) [reply ios/RNWalletModule/RNWalletModule.m]: tks, we'll change

Comment (jgfidelis) [reply ios/RNWalletModule/RNWalletModule.m]:
```
After further research into the source code I found that you can't declare an NSData argument, you have to pass it in as a base64 encoded string (NSString) and then use RCTConvert to convert it to an NSData.
```
```
RCT_EXPORT_METHOD(write:(CBUUID *)peripheralUuid 
                  forCharacteristicUuid:(CBUUID *)characteristicUuid 
                  data:(NSString *)data
                  withResponse:(BOOL)withResponse)
{
    // ....
    [peripheral 
          writeValue:[RCTConvert NSData:data] 
          forCharacteristic:/* ... */
          type:/* ... */];
}
```

this is easily fixable, we'll change it.

**Comment by sibelius** [comment]
one problem that we had, it that fetch is transform the .pkpass in a text and when the `data` goes to the ios bridge it marks the data as invalid

**Comment by erikpoort** [comment]
@sibelius As far as I know a pkpass is a zip, maybe that's the problem?

**Comment by sibelius** [comment]
@erikpoort we changed the pkpass from passing the pkpass data to passing a file uri

react-native will support blob on react-native 54, until then we used react-native-fetch-blob package to download .pkpass directly to device

then this package reads the content of filepath and share the pkpass, it worked well right now

let me know what do u think of this solution

**Comment by erikpoort** [comment]
Looks good, no time for me to test, but I'm guessing you have!

**Comment by sibelius** [comment]
tks

**Pull request merged**